lib/DiscoveryStreamFeed.jsm

```diff
@@ -534,6 +534,7 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
   }
 
   async disable() {
```

Comment:
if you're going to use this for both the disable and change case,  think it might be good to actually rename this function and add a comment saying it's used in both cases in case someone accidentally adds something that is disable-specific without realizing it's used when the pref is changed. So maybe `resetState` or something?

Reply:
I like `reset`, does that work?

Reply:
OK, I'm maybe we go with `resetState`, pull `clearCache` and `clearImpressionPrefs` out, rename them `resetCache` and `resetImpressionsPrefs`, and add a `reset` that calls those three.

```
  async reset() {
    this.resetImpressionPrefs();
    await this.resetCache();
    this.resetState();
  }

  async resetCache() {
    await this.cache.set("layout", {});
    await this.cache.set("feeds", {});
    await this.cache.set("spocs", {});
  }

  resetImpressionPrefs() {
    this.writeImpressionsPref(PREF_SPOC_IMPRESSIONS, {});
    this.writeImpressionsPref(PREF_REC_IMPRESSIONS, {});
  }

  resetState() {
    // Reset reducer
    this.store.dispatch(ac.BroadcastToContent({type: at.DISCOVERY_STREAM_LAYOUT_RESET}));
    this.loaded = false;
    this.layoutRequestTime = undefined;
    this.spocsRequestTime = undefined;
    this.componentFeedRequestTime = undefined;
    this.totalRequestTime = undefined;
  }

  async onPrefChange() {
    // We always want to clear the cache/state if the pref has changed
    await this.reset();
    if (this.config.enabled) {
      // Load data from all endpoints
      await this.enable();
    }
  }
```